### PR TITLE
Use grid and reduce duplicated code

### PR DESCRIPTION
**Pull request by RoyEJohnson**
Get rid of flexrow notion; use grid
Multiple `@include`s and the use of `@extend` caused a lot of duplication in the generated CSS file. It was about 167K before, and is now about 17K.

### CODE REVIEW

File: core/pattern-library/elements.scss [hunk 1]
```diff
@@ -1,6 +1,82 @@
...
+}
+
+// These should be @extend %text-h1, etc., but that causes SASS to duplicated
+// lots of stuff.
+h1 {
+    @include set-font(h1);
+}
+
+.hero h1 {
+    // @extend %text-hero-h1;
```
Comment (philschatz) [inline core/pattern-library/elements.scss]: Are these comments useful or can they be removed?

File: core/pattern-library/molecules.scss [hunk 2]
```diff
@@ -8,10 +8,39 @@ $form-border-width: 0.1rem;
...
     @include center;
     @include set-font(body-regular);
 
+    // Large button is the default
+    box-sizing: border-box;
+    cursor: pointer;
+    display: inline-flex;
+    height: $form-element-height;
+    margin: 0;
+    padding: 0 2 * $normal-margin;
```
Comment (philschatz) [inline core/pattern-library/molecules.scss]: Should the `2` have a unit?

File: core/pattern-library/molecules.scss [hunk 3]
```diff
@@ -48,36 +64,26 @@ $form-border-width: 0.1rem;
...
-    @include button(primary);
-}
-
-%button%secondary {
-    @include button(secondary);
+    @include button;
 }
 
-@mixin button-bar() {
+@mixin button-bar {
```
Comment (philschatz) [inline core/pattern-library/molecules.scss]: Why do some mixins with no args have `()` and others do not? What is the advantage to removing the `()`?

**Nit:** I find it easier when scanning to see `()` when a mixin is being called

File: core/pattern-library/elements.scss [hunk 4]
```diff
@@ -1,6 +1,82 @@
...
-@import 'elements/headings';
+
+html {
+    font-family: 'Neue Helvetica W01', sans-serif;
+    // This resets rem from 16px to 10px
+    font-size: 62.5%;
+}
+
+body {
+    // @extend %text-body-regular;
```
Comment (philschatz) [inline core/pattern-library/elements.scss]: Is this comment still needed?

File: core/pattern-library/molecules.scss (hunk 2 above)
Comment (RoyEJohnson) [reply core/pattern-library/molecules.scss]: No, `$normal-margin` has a unit.

File: core/pattern-library/elements.scss (hunk 1 above)
Comment (RoyEJohnson) [reply core/pattern-library/elements.scss]: I'll remove them.

File: core/pattern-library/molecules.scss (hunk 3 above)
Comment (RoyEJohnson) [reply core/pattern-library/molecules.scss]: I just discovered that empty parens are optional. :)

**Comment by RoyEJohnson** [comment]
I think it would be good to add a linter. I would do that as a separate check-in, as it will likely require a fair amount of revisions itself.

I'm not clear on how we would do code coverage with this package, or what HTML you're referring to.

File: core/pattern-library/molecules.scss (hunk 3 above)
Comment (RoyEJohnson) [reply core/pattern-library/molecules.scss]: Do you care about parens on mixin definitions?

File: script/build [hunk 5]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-node_modules/.bin/sass-lint -c sass-lint.yml -v -q
+node_modules/.bin/sass-lint -c sass-lint.yml -v -q || exit $?
```
Comment (philschatz) [inline script/build]: Ooh, you added an exit clause for this one. Should the other lines in this file also cause the build to fail when they do not run without errors?

File: core/pattern-library/molecules.scss (hunk 2 above)
Comment (philschatz) [reply core/pattern-library/molecules.scss]: oh I mis-parsed that expression. I thought you were setting the padding for top/bottom/left/right (4 items and left padding was `*` for some reason).

Instead, this expression says "Set the padding for top/bottom to be `0` and the padding for left/right to be `(2 * $normal-margin)`"

If I missed this maybe others would too?

**Comment by philschatz** [comment]
The code coverage could run on the HTML files that are generated by KSS. https://github.com/philschatz/css-coverage.js would take each HTML file and the CSS file and then output coverage data that is sent to codecov.

This is similar to what is done in https://github.com/Connexions/cnx-recipes (see any Pull Request for an example)

**Pull request merged**